staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -660,6 +710,47 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 	return nil
 }
 
+func (o *ApplyOptions) calculateSSAMigrationPatch(obj runtime.Object) ([]byte, error) {
```

Comment:
While this function is not exported, I would add at least some comments describing the patch created. And this function should have strong unit test coverage.

Reply:
While it's not common to unit test helper functions, in the case where they are particularly intricate or dense, we should. And this function has straightforward inputs/outputs that make unit testing easier. Setting up the fake `ApplyOptions` for the unit tests is accomplished in several other unit tests, including in the `cli-utils` repository: https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/apply/task/apply_task.go#L188

Reply:
Is this function super dense or intricate? I'm sure sure unit testing it specifically has much value especially given the `UpgradeManagedFields` is already well tested. 

The only additional logic unit tests for this function would add coverage for is the `jsonPatch` construction - and I'm imagining that test would basically just have this same code?